guides/source/testing.md

```diff
@@ -1242,378 +1375,320 @@ assert_response :redirect
---- expected
-+++ actual
-@@ -1 +1 @@
--"Article was successfully created."
-+nil
+It should have created a test file placeholder for us. With the output of the
+previous command you should see:
 
-1 runs, 4 assertions, 1 failures, 0 errors, 0 skips
+```
+      invoke  test_unit
+      create    test/system/articles_test.rb
 ```
 
-Let's implement the flash message now in our controller. Our `:create` action should now look like this:
+Now let's open that file and write our first assertion:
 
 ```ruby
-def create
-  @article = Article.new(article_params)
+require "application_system_test_case"
 
-  if @article.save
-    flash[:notice] = "Article was successfully created."
-    redirect_to @article
-  else
-    render "new"
+class ArticlesTest < ApplicationSystemTestCase
+  test "viewing the index" do
+    visit articles_path
+    assert_selector "h1", text: "Articles"
   end
 end
 ```
 
-Now if we run our tests, we should see it pass:
+The test should see that there is an `h1` on the articles index page and pass.
+
+Run the system tests.
 
 ```bash
-$ bin/rails test test/controllers/articles_controller_test.rb -n test_should_create_article
-Run options: -n test_should_create_article --seed 18981
+$ bin/rails test:system
+```
 
-# Running:
+NOTE: By default, running `bin/rails test` won't run your system tests. Make
+sure to run `bin/rails test:system` to actually run them. You can also run
+`bin/rails test:all` to run all tests, including system tests.
 
-.
+#### Creating Articles System Test
 
-Finished in 0.081972s, 12.1993 runs/s, 48.7972 assertions/s.
+Now let's test the flow for creating a new article in our blog.
 
-1 runs, 4 assertions, 0 failures, 0 errors, 0 skips
-```
+```ruby
+test "should create Article" do
+  visit articles_path
 
-### Putting It Together
+  click_on "New Article"
 
-At this point our Articles controller tests the `:index` as well as `:new` and `:create` actions. What about dealing with existing data?
+  fill_in "Title", with: "Creating an Article"
+  fill_in "Body", with: "Created this article successfully!"
 
-Let's write a test for the `:show` action:
+  click_on "Create Article"
 
-```ruby
-test "should show article" do
-  article = articles(:one)
-  get article_url(article)
-  assert_response :success
+  assert_text "Creating an Article"
 end
 ```
 
-Remember from our discussion earlier on fixtures, the `articles()` method will give us access to our Articles fixtures.
+The first step is to call `visit articles_path`. This will take the test to the
+articles index page.
+
+Then the `click_on "New Article"` will find the "New Article" button on the
+index page. This will redirect the browser to `/articles/new`.
 
-How about deleting an existing Article?
+Then the test will fill in the title and body of the article with the specified
+text. Once the fields are filled in, "Create Article" is clicked on which will
+send a POST request to create the new article in the database.
```

Comment:
Since we don't talk about "the database" anywhere else, maybe a little odd to see it here?

```suggestion
text. Once the fields are filled in, "Create Article" is clicked on which will
send a POST request to `/articles/create`.
```